packages/hash/api/src/model/knowledge/linkEntity.model.ts

```diff
@@ -0,0 +1,187 @@
+import {
+  GraphApi,
+  Entity,
+  LinkEntityMetadata,
+  Filter,
+  EntityStructuralQuery,
+} from "@hashintel/hash-graph-client";
+
+import { EntityModel, EntityTypeModel, LinkEntityModel } from "../index";
+
+export type LinkModelConstructorParams = {
+  linkEntity: Entity;
+  linkEntityTypeModel: EntityTypeModel;
+  leftEntityModel: EntityModel;
+  rightEntityModel: EntityModel;
+};
+
+export type LinkModelCreateParams = {
+  ownedById: string;
+  linkEntityTypeModel: EntityTypeModel;
+  leftEntityModel: EntityModel;
+  leftOrder?: number;
+  rightEntityModel: EntityModel;
+  rightOrder?: number;
+  actorId: string;
+};
+
+/**
+ * @class {@link LinkEntityModel}
+ */
+export default class extends EntityModel {
+  leftEntityModel: EntityModel;
```

Comment:
Should these be protected?

Reply:
They weren't protected previously, and they are currently accessed directly outside of the model class - so for now I don't think so.